### ISSUE TITLE
Ubuntu 20.04 and Fedora 32 build results

### ISSUE THREAD

**Issue by sdarwin** [boostorg/beast#2029]
### Version of Beast

299
### Steps necessary to reproduce the problem

(from beast/pull/2028)

docker build -f .dockers/ubuntu-20/Dockerfile .

### Summary of result

```
...failed updating 2 targets...
...updated 4200 targets...

libs/beast/test/beast/core/basic_stream.cpp:1346:20: error: no matching member function for call to 'async_connect'
```
### Further output

```
clang-linux.compile.c++.without-pch bin.v2/libs/beast/test/beast/core/basic_stream.test/clang-linux-10.0.0/release/cxxstd-2a-iso/threadapi-pthread/threading-multi/visibility-hidden/basic_stream.o
In file included from libs/beast/test/beast/core/basic_stream.cpp:26:
In file included from ./boost/asio/spawn.hpp:19:
In file included from ./boost/coroutine/all.hpp:11:
In file included from ./boost/coroutine/coroutine.hpp:10:
In file included from ./boost/coroutine/asymmetric_coroutine.hpp:24:
In file included from ./boost/coroutine/exceptions.hpp:14:
 ./boost/detail/scoped_enum_emulation.hpp:17:1: warning: This header is deprecated. Use <boost/core/scoped_enum.hpp> instead. [-W#pragma-messages]
BOOST_HEADER_DEPRECATED("<boost/core/scoped_enum.hpp>")
^
./boost/config/header_deprecated.hpp:23:37: note: expanded from macro 'BOOST_HEADER_DEPRECATED'
# define BOOST_HEADER_DEPRECATED(a) BOOST_PRAGMA_MESSAGE("This header is deprecated. Use " a " instead.")
                                  ^
./boost/config/pragma_message.hpp:24:34: note: expanded from macro 'BOOST_PRAGMA_MESSAGE'
# define BOOST_PRAGMA_MESSAGE(x) _Pragma(BOOST_STRINGIZE(message(x)))
                             ^
 <scratch space>:133:2: note: expanded from here
message("This header is deprecated. Use " "<boost/core/scoped_enum.hpp>" " instead.")
^
libs/beast/test/beast/core/basic_stream.cpp:1346:20: error: no matching member function for call to 'async_connect'
             stream.async_connect(
             ~~~~~~~^~~~~~~~~~~~~
./boost/beast/core/basic_stream.hpp:1203:5: note: candidate template ignored: constraints not satisfied [with Iterator = boost::asio::ip::basic_resolver_iterat     or<boost::asio::ip::tcp>, ConnectCondition = condition, IteratorConnectHandler = handler]
    async_connect(
    ^
./boost/beast/core/basic_stream.hpp:1196:9: note: because '::boost::asio::completion_token_for<handler, void (boost::beast::error_code, boost::asio::ip::basic_     resolver_iterator<boost::asio::ip::tcp>)>' evaluated to false
         BOOST_ASIO_COMPLETION_TOKEN_FOR(
       ^
./boost/asio/async_result.hpp:502:18: note: expanded from macro 'BOOST_ASIO_COMPLETION_TOKEN_FOR'
::boost::asio::completion_token_for<s>

**Comment by madmongo1** [comment]
I Strongly suspect this is because gcc-9's concepts support is incomplete. Checking.

**Comment by madmongo1** [comment]
This is an asio issue. I'll add a workaround to the dockerfile

**Comment by madmongo1** [comment]
Turns out that this was a problem in the tests. Fixed in 331a38755
